src/m365/pp/commands/card/card-get.spec.ts

```diff
@@ -0,0 +1,196 @@
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it('defines correct properties for the default output', () => {
+    assert.deepStrictEqual(command.defaultProperties(), ['name', 'cardid', 'publishdate', 'createdon', 'modifiedon']);
+  });
+
+  it('retrieves specific card', async () => {
+    sinon.stub(request, 'get').callsFake(async opts => {
+      if ((opts.url === `https://api.bap.microsoft.com/providers/Microsoft.BusinessAppPlatform/environments/4be50206-9576-4237-8b17-38d8aadfaa36?api-version=2020-10-01&$select=properties.linkedEnvironmentMetadata.instanceApiUrl`)) {
+        if (opts.headers &&
+          opts.headers.accept &&
+          (opts.headers.accept as string).indexOf('application/json') === 0) {
+          return envResponse;
+        }
+      }
+
+      if ((opts.url === `https://contoso-dev.api.crm4.dynamics.com/api/data/v9.1/cards(3a081d91-5ea8-40a7-8ac9-abbaa3fcb893)`)) {
+        if (opts.headers &&
+          opts.headers.accept &&
+          (opts.headers.accept as string).indexOf('application/json') === 0) {
+          return cardResponse;
+        }
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, { options: { debug: true, environment: '4be50206-9576-4237-8b17-38d8aadfaa36', id: '3a081d91-5ea8-40a7-8ac9-abbaa3fcb893' } });
+    assert(loggerLogSpy.calledWith(cardResponse));
+
+  });
+
+  it('retrieves cards as admin', async () => {
+    sinon.stub(request, 'get').callsFake(async opts => {
+      if ((opts.url === `https://api.bap.microsoft.com/providers/Microsoft.BusinessAppPlatform/scopes/admin/environments/4be50206-9576-4237-8b17-38d8aadfaa36?api-version=2020-10-01&$select=properties.linkedEnvironmentMetadata.instanceApiUrl`)) {
+        if (opts.headers &&
+          opts.headers.accept &&
+          (opts.headers.accept as string).indexOf('application/json') === 0) {
+          return envResponse;
+        }
+      }
+
+      if ((opts.url === `https://contoso-dev.api.crm4.dynamics.com/api/data/v9.1/cards(3a081d91-5ea8-40a7-8ac9-abbaa3fcb893)`)) {
+        if (opts.headers &&
+          opts.headers.accept &&
+          (opts.headers.accept as string).indexOf('application/json') === 0) {
+          return cardResponse;
+        }
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, { options: { debug: true, environment: '4be50206-9576-4237-8b17-38d8aadfaa36', id: '3a081d91-5ea8-40a7-8ac9-abbaa3fcb893', asAdmin: true } });
+    assert(loggerLogSpy.calledWith(cardResponse));
+  });
+
+  it('correctly handles API OData error', async () => {
+    sinon.stub(request, 'get').callsFake(async opts => {
+      if ((opts.url === `https://api.bap.microsoft.com/providers/Microsoft.BusinessAppPlatform/environments/4be50206-9576-4237-8b17-38d8aadfaa36?api-version=2020-10-01&$select=properties.linkedEnvironmentMetadata.instanceApiUrl`)) {
```

Comment:
Could we let the actual retrieval of the card fail instead of the retrieval of the environment?

Reply:
i think not, let me know if i'm wrong. I removed the retrievel of the environment since it makes no sense here